crates/net/discv4/src/lib.rs

```diff
@@ -1144,8 +1164,7 @@ impl Discv4Service {
             return
         }
 
-        let key = kad_key(id);
-        if let BucketEntry::Present(_, _) = self.kbuckets.entry(&key) {
+        if self.has_bond(id, remote_addr.ip()) {
```

Comment:
Should this still include the old condition? Or just do that if we have a bond, and it's not possible to not have a bond and be present in the bucket?

Reply:
enforcing bond here is sufficient